src/direct_mlj.jl

```diff
@@ -47,7 +47,25 @@ end
 Laplace_Models = Union{LaplaceRegressor,LaplaceClassifier}
 
 # for fit:
-MMI.reformat(::Laplace_Models, X, y) = (MLJBase.matrix(X) |> permutedims, reshape(y, 1, :))
+MMI.reformat(::LaplaceRegressor, X, y) = (MLJBase.matrix(X) |> permutedims, (reshape(y, 1, :),nothing))
+
+
+function MMI.reformat(::LaplaceClassifier, X, y) 
+
+    X = MLJBase.matrix(X) |> permutedims
+
+
+    y= reshape(y, 1, :)
+    # Convert labels to integer format starting from 0 for one-hot encoding
+    y_plain = MLJBase.int(y[1, :]) .- 1
+    # One-hot encoding of labels
+    unique_labels = unique(y_plain) # Ensure unique labels for one-hot encoding
+    y_onehot = Flux.onehotbatch(y_plain, unique_labels) # One-hot encoding
+    return X,(y_onehot, y[1])
+end
+
+#MMI.selectrows(::LaplaceClassifier, I, Xmatrix, y) = (view(Xmatrix, :, I), (view(y[1],I),y[2]))
```

Comment:
@pat-alt  these two lines  commentend out are the last big problem. I was able to remove the if statements from the fit! and update functions but  I encounter a shape issue when  i tried to implement them and i couldn't make them work correctly. Without these two lines ```evaluate!(mach, resampling=cv, measure=l2, verbosity=0)``` doesn't work. aside from this, everything else seems to work.

Reply:
OK, I'll have a look at this now, thanks